context/ethereumContext.tsx

```diff
@@ -478,26 +514,43 @@ export const EthereumProvider: React.FC<{}> = ({ children }) => {
     setPrecompiled(precompiled)
   }
 
-  const _setupStateManager = () => {
-    // Hack the state manager so that we can track the stored variables
-    // @ts-ignore: Store original contract storage to access later
-    vm.stateManager.originalPutContractStorage =
-      vm.stateManager.putContractStorage
+  function traceMethodCalls(obj: any) {
+    const handler = {
+      get(target: any, propKey: any, receiver: any) {
+        const origMethod = target[propKey]
+        return (...args: any[]) => {
+          const result = origMethod.apply(target, args)
+          if (propKey == 'clearContractStorage') {
+            _clearContractStorage(args[0])
+          }
+          if (propKey == 'putContractStorage') {
+            _putContractStorage(args[0], args[1], args[2])
+          }
+          return result
+        }
+      },
+    }
+    return new Proxy(obj, handler)
+  }
 
-    // @ts-ignore: Store original contract storage to access later
-    vm.stateManager.originalClearContractStorage =
-      vm.stateManager.clearContractStorage
+  // In this function we create a proxy EEI object that will intercept
+  // putContractStorage and clearContractStorage and route them to our
+  // implementations at _putContractStorage and _clearContractStorage
+  // respectively AFTER applying the original methods.
+  // This is necessary in order to handle storage operations easily.
+  const _setupStateManager = () => {
+    var proxyStateManager = traceMethodCalls(vm.evm.eei)
+    vm.evm.eei.putContractStorage = proxyStateManager.putContractStorage
+    vm.evm.eei.clearContractStorage = proxyStateManager.clearContractStorage
 
-    vm.stateManager.putContractStorage = _putContractStorage
-    vm.stateManager.clearContractStorage = _clearContractStorage
     storageMemory.clear()
   }
 
   const _setupAccount = () => {
     // Add a fake account
     const accountData = {
       nonce: 2,
-      balance: new BN(10).pow(new BN(accountBalance)),
+      balance: BigInt(10 ** accountBalance),
```

Comment:
```suggestion
      balance: 10n ** BigInt(accountBalance),
```

Reply:
I'm getting a runtime error when committing this change:
`TypeError: Cannot convert a BigInt value to a number`

Reply:
I see. It compiles it into call `Math.pow(10n, BigInt(accountBalance))`, and `Math.pow` expects `number`.